doc/explanation/package-management.md

```diff
@@ -0,0 +1,220 @@
+* Per-project dependencies
+
+Dune plays well with the existing OCaml ecosystem and does not introduce a new
+type of packages - it uses the same package repository and Dune packages stay
+installable with OPAM.
+
+## Package Management in a Project
+
+This section describes what happens in a Dune project that is using the package
+management feature.
+
+## Dependency Selection
+
+The first step is to determine which packages need to be installed.
+Traditionally this has been defined in the `depends` field of a projects OPAM
+file(s).
+
+Since a while Dune has also supported {doc}`OPAM file generation
+</howto/opam-file-generation>` by specifying the package dependencies in the
+`dune-project`. Outside of this feautre, the `depends` stanza had not been used
+by Dune.
+
+The package management feature changes this, as Dune now determines the list of
+packages to install from the `depends` stanza in the `dune-project` file. This
+allows projects to completely omit generation of `.opam` files, as long as they
+use Dune for package management. Thus all dependencies are only declared in one
+file.
+
+For compatibility with a larger amount of existing projects Dune will also
+collect dependencies from `.opam` files in the project. So while recommended,
+there is no obligation to switch to declaring dependencies in the
+`dune-project`. Likewise the generation of `.opam` files will still work.
+
+## Locking
+
+To go from a set of dependency constraints for a project to a set of installed
+packages and versions there needs to be a step to determine the right packages
+and their versions to be installed.
+
+In `opam` this process happens as part of `opam install`, thus linking finding
+a solution that satifies the given constrains and installation into one step.
+Dune on the other hand moves finding a solution and installing into separate
+steps. First a solution is found and then packages are installed separately.
+The idea of finding a solution and recording it for later is popular in many
+package managers and is usually called locking.
+
+:::{note}
+`opam` also supports creating lock-files. However these are not as central to
+the opam workflow as they are in the case of package management in Dune, which
+always requires a set of locked packages.
+:::
+
+A package lock is in the most general sense just a set of specific packages and
+their versions to be installed.
+
+A Dune lockfile extends this to a directory with files that describe the
+dependencies. It includes the name of the package and the version. Unlike many
+other package managers, the files include a lot of other information as well,
+such as the location of the source archives to download (since there is no
+central location for all archives), the build instructions (since each package
+can use its own way of building) as well as additional metadata like the system
+packages it depends upon.
+
+The files are stored in a directory as separate files, because that makes it
```

Comment:
It could be interesting to mention the `dune.lock` dir name.

Reply:
I was a bit hesitant as `dune.lock` feels a bit like an implementation detail but I guess in case we change it we can change this document as well.